app/src/main/java/org/mozilla/fenix/settings/SignOutFragment.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.coroutines.launch
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.requireComponents
+
+class SignOutFragment : BottomSheetDialogFragment() {
+    private lateinit var accountManager: FxaAccountManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(DialogFragment.STYLE_NO_TITLE, R.style.FirefoxAccountsDialogStyle)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        accountManager = requireComponents.backgroundServices.accountManager
+        return inflater.inflate(R.layout.fragment_sign_out, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val signOut = view.findViewById(R.id.sign_out_disconnect) as Button
+        signOut.setOnClickListener {
```

Comment:
Performance nit: One could reference `view.signOut` to use Kotlin synthetic methods and a View Cache. This offers a very minor performance improvement by reducing the number of view hierarchy passes, which is strictly optional given that it can have a code safety impact if the view goes away by then. Of course, casting the result of findViewById as non-Nullable has the same code safety issue.